src/main/java/com/chetanmeh/oak/index/config/generator/FunctionNameConverter.java

```diff
@@ -0,0 +1,65 @@
+package com.chetanmeh.oak.index.config.generator;
+
+import java.util.List;
+
+public class FunctionNameConverter {
+  public final static List<String> functionNames = List.of("local-name", "name", "path", "upper-case", "lower-case",
+    "coalesce", "first", "string-length");
+
+  public static String apply(String functionPattern) {
+    String functionName = extractFunctionName(functionPattern);
```

Comment:
This won't work for 
* nested functions, e.g. `upper(lower(@test))`
* SQL-2 functions
* functions with multiple properties (e.g. coalesce)

Can't we use the existing parser logic in some way instead?

It seems it would be much simpler, and avoid all those problems, if we use the polish notation instead:
* split by "*"
* iterate over the entries
* each property starts with "@"

Reply:
Yes this was just to sketch out the initial idea! Was writing that already! Almost finished the implementation for it.